src/biconnected.jl

```diff
@@ -0,0 +1,188 @@
+"""
+
+biconnected Component Decomposition
+----------------------------------
+
+Any connected graph decomposes into a tree of biconnected components 
+called the block-cut tree of the graph. The blocks are attached to each 
+other at shared vertices called cut vertices or articulation points.
+
+This implementation is based on the algorithm provided by Tarjan 
+in "Depth-First Search and Linear Graph Algorithms".  
+
+
+'1' in the entry of the articulation point array denotes that the vertex is an articulation point.
+
+
+Functions
+---------
+
+biconnected_component(A::MatrixNetwork)                         
+Returns an array that represents the biconnected component number for the corrsponding 
+edge in its csr representation.
+
+enrich_biconnected(A::MatrixNetwork, biconnected_component_id::Int64)  
+Returns the largest biconnected component if the biconnected_component_id parameter is zero or
+less. Else return the component specified.  
+
+
+"""
+
+type biconnected_components_output
+    biconnected_components_label::Vector{Int64} #biconnected_component_number
+    articulation_points::Vector{Int64}  
+    A::MatrixNetwork #MatrixNetwork
+end
+
+
+
+function biconnected_component(A::MatrixNetwork)
```

Comment:
You should have inputs for sparse matrices too that auto-convert as done in the other functions. Also, probably best to have a biconnected_component_map function that just gives componetns and an "articulation_points" function that just returns the articulation points.